cpp/tests/community/edge_triangle_count_test.cpp

```diff
@@ -0,0 +1,260 @@
+      cugraph::decompress_to_edgelist(
+        handle,
+        graph_view,
+        std::optional<cugraph::edge_property_view_t<edge_t, weight_t const*>>{std::nullopt},
+        std::make_optional(d_cugraph_results.view()),
+        std::optional<raft::device_span<vertex_t const>>{std::nullopt});  // FIXME: No longer needed
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      hr_timer.start("EdgeTriangleCount");
+    }
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      hr_timer.stop();
+      hr_timer.display_and_clear(std::cout);
+    }
+
+    if (edge_triangle_count_usecase.check_correctness_) {
+      std::optional<cugraph::graph_t<vertex_t, edge_t, false, false>> modified_graph{std::nullopt};
+      std::vector<vertex_t> h_srcs(edgelist_srcs.size());
+      std::vector<vertex_t> h_dsts(edgelist_dsts.size());
+      std::tie(h_srcs, h_dsts, std::ignore) = cugraph::test::graph_to_host_coo(
+        handle,
+        graph_view,
+        edge_weight ? std::make_optional((*edge_weight).view()) : std::nullopt,
+        std::optional<raft::device_span<vertex_t const>>(std::nullopt));
+
+      auto h_cugraph_edge_triangle_counts = cugraph::test::to_host(handle, *d_edge_triangle_counts);
+
+      auto h_reference_edge_triangle_counts =
+        edge_triangle_count_reference<vertex_t, edge_t>(h_srcs, h_dsts);
+
+      for (size_t i = 0; i < h_srcs.size(); ++i) {
+        ASSERT_EQ(h_cugraph_edge_triangle_counts[i], h_reference_edge_triangle_counts[i])
+          << "Edge triangle count values do not match with the reference values.";
+      }
+    }
+  }
+};
+
+using Tests_EdgeTriangleCount_File = Tests_EdgeTriangleCount<cugraph::test::File_Usecase>;
+using Tests_EdgeTriangleCount_Rmat = Tests_EdgeTriangleCount<cugraph::test::Rmat_Usecase>;
+
+TEST_P(Tests_EdgeTriangleCount_File, CheckInt32Int32Float)
+{
+  run_current_test<int32_t, int32_t, float>(
+    override_File_Usecase_with_cmd_line_arguments(GetParam()));
+}
+TEST_P(Tests_EdgeTriangleCount_Rmat, CheckInt32Int32Float)
+{
+  run_current_test<int32_t, int32_t, float>(
+    override_Rmat_Usecase_with_cmd_line_arguments(GetParam()));
+}
+TEST_P(Tests_EdgeTriangleCount_File, CheckInt64Int64Float)
+{
+  run_current_test<int64_t, int64_t, float>(
+    override_File_Usecase_with_cmd_line_arguments(GetParam()));
+}
+TEST_P(Tests_EdgeTriangleCount_Rmat, CheckInt64Int64Float)
+{
+  run_current_test<int64_t, int64_t, float>(
+    override_Rmat_Usecase_with_cmd_line_arguments(GetParam()));
+}
+
+INSTANTIATE_TEST_SUITE_P(
+  simple_test,
+  Tests_EdgeTriangleCount_File,
+  ::testing::Combine(
+    // enable correctness checks
+    ::testing::Values(EdgeTriangleCount_Usecase{false, false, true},
+                      EdgeTriangleCount_Usecase{
+                        true, false, true}),  // FIXME: Still debugging edge_mask
```

Comment:
I think we don't need test_weighted here. Weights have no role in TC. Need to include both edge_mask = false. and true  simple_test, rmat_small_test, and rmat_benchmark.